crates/ruff_linter/src/rules/pylint/rules/misplaced_bare_raise.rs

```diff
@@ -0,0 +1,100 @@
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::statement_visitor::StatementVisitor;
+use ruff_python_ast::{statement_visitor, Stmt};
+use ruff_python_semantic::NodeRef;
+use ruff_text_size::Ranged;
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+///
+/// This rule triggers an error when a bare raise statement is not in an except or finally block.
+///
+/// ## Why is this bad?
+///
+/// If raise statement is not in an except or finally block, there is no active exception to
+/// re-raise, so it will fail with a `RuntimeError` exception.
+///
+/// ## Example
+/// ```python
+/// def validate_positive(x):
+///     if x <= 0:
+///         raise
+/// ```
+///
+/// Use instead:
+/// ```python
+/// def validate_positive(x):
+///     if x <= 0:
+///         raise ValueError(f"{x} is not positive")
+/// ```
+#[violation]
+pub struct MisplacedBareRaise;
+
+impl Violation for MisplacedBareRaise {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("The raise statement is not inside an except clause")
+    }
+}
+
+#[derive(Default)]
+struct RaiseFinder<'a> {
+    raises: Vec<&'a Stmt>,
+}
+
+impl<'a, 'b> StatementVisitor<'b> for RaiseFinder<'a>
+where
+    'b: 'a,
+{
+    fn visit_stmt(&mut self, stmt: &'b Stmt) {
+        match stmt {
+            Stmt::Raise(_) => self.raises.push(stmt),
+            _ => statement_visitor::walk_stmt(self, stmt),
+        }
+    }
+}
+
```

Comment:
I wanted to use [`ruff_python_ast::helpers::RaiseStatementVisitor`](https://github.com/astral-sh/ruff/blob/main/crates/ruff_python_ast/src/helpers.rs#L895), but it doesn't store the raise statement itself.

It's only used in two rules, so maybe I can refactor that?

Reply:
Ah yeah -- can we change `RaiseStatementVisitor` to store `Vec<&'a ast::StmtRaise>`?

Reply:
Shall I do that in a separate PR?

Reply:
I would be cool with including it here, but either is fine.

Reply:
Not needed anymore, so I guess I won't touch `RaiseStatementVisitor`.